File: Data Structures & Algorithms/largest-rectangle-in-histogram/submission-3.py
class Solution:
    def largestRectangleArea(self, heights: List[int]) -> int:
        n = len(heights)
        st = []
        res = 0

        for i in range(n):
            start = i
            while st and heights[i] < st[-1][1]:
                start = st[-1][0]
                res = max(res, st[-1][1] * (i - st[-1][0]))
                st.pop()

            st.append((start, heights[i]))
        
        while st:
            res = max(res, st[-1][1] * (n-st[-1][0]))
            st.pop()

        return res

File: Data Structures & Algorithms/is-anagram/submission-3.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        int n = s.size(), m = t.size();

        vector<int> c(26, 0);

        for(int i=0; i<n; i++){
            c[s[i]-'a']++;
        }

        for(int i=0; i<m; i++){
            c[t[i]-'a']--;
        }

        for(int i=0; i<26; i++){
            if(c[i])
                return false;
        }

        return true;
    }
};
